code/game/turfs/open/floor/catwalk_plating.dm

```diff
@@ -16,24 +16,28 @@
 	overfloor_placed = TRUE
 	underfloor_accessibility = UNDERFLOOR_VISIBLE
 	var/covered = TRUE
+	var/above_state = "maint_above"	//Icon-state for the overlay
 
 
 /turf/open/floor/catwalk_floor/Initialize(mapload)
 	. = ..()
 	update_icon(UPDATE_OVERLAYS)
 
+GLOBAL_LIST_INIT(catwalk_overlay_masterlist, list())	//Stores all the above_states for the different types of catwalk
+
 /turf/open/floor/catwalk_floor/update_overlays()
 	. = ..()
-	var/static/image/catwalk_overlay
-	if(isnull(catwalk_overlay))
-		catwalk_overlay = new()
-		catwalk_overlay.icon = icon
-		catwalk_overlay.icon_state = "catwalk_above"
-		catwalk_overlay.plane = GAME_PLANE
-		catwalk_overlay.layer = CATWALK_LAYER
-		catwalk_overlay = catwalk_overlay.appearance
 	if(covered)
```

Comment:
`if(covered)` should be replaced with `if(!covered) ...(next line) return` so we don't indent too deep

Reply:
This gives a good explanation as to why https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#use-early-returns

Reply:
I felt sick coding that line, this is why, thanks for calling me out LMAO